src/services/api/user-api.ts

```diff
@@ -0,0 +1,45 @@
+import { HTTPTransport } from '../../core'
+
+export interface IChangePasswordData {
+  oldPassword: string
+  newPassword: string
+}
+
+interface IChangeUserData {
+  first_name: string
+  second_name: string
+  display_name: string
+  login: string
+  email: string
+  phone: string
+}
+
+export default class UserApi {
+  http: HTTPTransport
+
+  constructor(http: HTTPTransport) {
+    this.http = http
+  }
+
+  async changePassword(data: IChangePasswordData): Promise<XMLHttpRequest> {
+    const res = await this.http.put('user/password', {
+      data: { oldPassword: data.oldPassword, newPassword: data.newPassword },
+    })
+    return res
```

Comment:
```suggestion
    return this.http.put('user/password', {
      data: { oldPassword: data.oldPassword, newPassword: data.newPassword },
    })
```

Можно более кратко записывать, если нет никаких промежуточных преобразований